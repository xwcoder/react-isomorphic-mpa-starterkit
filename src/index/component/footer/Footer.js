import React from 'react'

//   fetchStatus: 0 // 0:未加载, 1:加载中, 2:加载成功, 3:加载失败，4: no more data
export default function Footer ({ fetchStatus }) {

  let footer = null

  if (fetchStatus === 1 || fetchStatus === 0 || fetchStatus === 2) {
    footer = <div className="app-footer">loading...</div>
  } else if (fetchStatus === 3) {
    footer = <div className="app-footer">加载失败，请刷新重试</div>
  }

  return footer
}
