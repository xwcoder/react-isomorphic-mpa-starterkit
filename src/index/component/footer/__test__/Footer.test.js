import React from 'react'
import { shallow } from 'enzyme'
import Footer from '../Footer'

describe('index/comonent/Footer', () => {

  it('footer with fetchStatus 0', () => {
    const wrapper = shallow(<Footer fetchStatus={0} />)
    expect(wrapper.find('.app-footer')).toHaveLength(1)
    expect(wrapper.text()).toEqual('loading...')
    expect(wrapper).toMatchSnapshot()
  })

  it('footer with fetchStatus 1', () => {
    const wrapper = shallow(<Footer fetchStatus={1} />)
    expect(wrapper.find('.app-footer')).toHaveLength(1)
    expect(wrapper.text()).toEqual('loading...')
    expect(wrapper).toMatchSnapshot()
  })

  it('footer with fetchStatus 2', () => {
    const wrapper = shallow(<Footer fetchStatus={2} />)
    expect(wrapper.find('.app-footer')).toHaveLength(1)
    expect(wrapper.text()).toEqual('loading...')
    expect(wrapper).toMatchSnapshot()
  })

  it('footer with fetchStatus 3', () => {
    const wrapper = shallow(<Footer fetchStatus={3} />)
    expect(wrapper.find('.app-footer')).toHaveLength(1)
    expect(wrapper.text()).toEqual('加载失败，请刷新重试')
    expect(wrapper).toMatchSnapshot()
  })

  it('footer with fetchStatus 4', () => {
    const wrapper = shallow(<Footer fetchStatus={4} />)
    expect(wrapper.find('.app-footer')).toHaveLength(0)
    expect(wrapper.text()).toEqual('')
    expect(wrapper).toMatchSnapshot()
  })
})
