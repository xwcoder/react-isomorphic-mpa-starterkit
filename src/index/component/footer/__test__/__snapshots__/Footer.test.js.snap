// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index/comonent/Footer footer with fetchStatus 0 1`] = `
<div
  className="app-footer"
>
  loading...
</div>
`;

exports[`index/comonent/Footer footer with fetchStatus 1 1`] = `
<div
  className="app-footer"
>
  loading...
</div>
`;

exports[`index/comonent/Footer footer with fetchStatus 2 1`] = `
<div
  className="app-footer"
>
  loading...
</div>
`;

exports[`index/comonent/Footer footer with fetchStatus 3 1`] = `
<div
  className="app-footer"
>
  加载失败，请刷新重试
</div>
`;

exports[`index/comonent/Footer footer with fetchStatus 4 1`] = `""`;
